review/core/interactors/authorization.py

```diff
@@ -0,0 +1,91 @@
+from core.enums import Phase
+from core.interactors.settings import is_at_phase, get_active_round
+from core.models import ProjectReview
+
+
+def can_participate(user, review_round):
+    return user in review_round.participants.all()
+
+
+def can_review_person(user, reviewee):
+    if not user.is_authenticated:
+        return False
+
+    if not can_participate(user, get_active_round()) or not can_participate(reviewee, get_active_round()):
+        return False
+
+    if is_at_phase(Phase.SELF_REVIEW):
+        if user == reviewee:
+            return True
+        return False
+
+    if is_at_phase(Phase.PEER_REVIEW):
+        if ProjectReview.objects.filter(reviewee=reviewee, reviewers=user).exists():
+            return True
+        return False
+    return False
+
+
+def can_review_project(user, project):
+    if not user.is_authenticated:
+        return False
+
+    if not can_participate(user, get_active_round()):
+        return False
+
+    if project not in get_active_round().projects.all():
```

Comment:
extract to a new function

Reply:
Which part exactly? I can imagine an `is_project_active` method. Do you mean something different?

Reply:
yes `is_project_active`